src/components/AvailabilityPageSingle.js

```diff
@@ -64,34 +103,29 @@ const AvailabilityPageSingle = ({ offer, error, useChangeStatus }) => {
           </Alert>
         ) : (
           [
+            offer.calendarType === CalendarType.SINGLE && (
+              <BookingAvailabilityForm
+                key="booking-availability"
+                bookingAvailabilityType={bookingAvailabilityType}
+                onChangeBookingAvailability={(e) =>
+                  setBookingAvailabilityType(e.target.value)
+                }
+              />
+            ),
             <StatusForm
               key="reason-and-type"
               offerType={offerType}
               statusType={type}
-              statusReason={reason}
+              statusReason={reasonMainLanguage}
               onChangeStatusType={(e) => setType(e.target.value)}
-              onInputStatusReason={(e) => setReason(e.target.value)}
+              onInputStatusReason={(e) => setReasonMainLanguage(e.target.value)}
             />,
             <Inline key="actions" spacing={3}>
               <Button
                 variant={ButtonVariants.PRIMARY}
-                disabled={!offer || reason.length > 200}
+                disabled={!offer || reasonMainLanguage.length > 200}
                 onClick={() => {
-                  if (type === OfferStatus.AVAILABLE) {
-                    changeStatusMutation.mutate({
-                      id: offerId,
-                      type,
-                    });
-                  } else {
-                    changeStatusMutation.mutate({
-                      id: offerId,
-                      type,
-                      reason: {
-                        ...(offer.status.type === type && offer.status.reason),
-                        [i18n.language]: reason || undefined,
-                      },
-                    });
-                  }
+                  changeStatusMutation.mutate(createMutationPayload());
```

Comment:
When I test this, the reason doesn't seem to get properly set

Reply:
There is an edge case when setting the status with a reason on an event with single calendar type. In that case the reason is only present on the sub event and not on the top level. The UI only uses the top level to get the status from a single calendar type.

Proposed solution: inside the backend also apply the status reason from the sub event on the top level.

Reply:
This PR fixes it https://github.com/cultuurnet/udb3-silex/pull/733

Reply:
@simon-debruijn I guess it's fine to already merge this now that the edge case is clear?

Reply:
Ok ✅